### ISSUE TITLE
Request "open-on-focus" attribute on datepicker-popup.

### ISSUE THREAD

**Issue by OlmoAcerta** [angular-ui/bootstrap#2303]
After upgrading to 0.11 our datepicker popups no longer work as expected. I've asked around on IRC and came to the conclusion that there are very good reasons for this (w.r.t. accessibility). 

Nevertheless, the guys from business prefer the old behavior. As they are the end-user, there is little I can do to convince them that the current behavior is fine.

As a compromise I propose an "open-on-focus" attribute, which could be put on datepicker-popups to re-enable the old behavior. A note should be made in the documentation to explain the accessibility issues involved. 

Thank you.

**Comment by jwanglof** [comment]
+1

**Comment by arjunasuresh3** [comment]
+1

**Comment by artdias90** [comment]
+1

**Comment by artdias90** [comment]
What's the status regarding closing datepicker on blur?

**Comment by sistemanick** [comment]
+1

**Comment by bademux** [comment]
looks like *onOpenFocus* doesn't work http://plnkr.co/edit/s15jAd1hEEI94dCeC42I?p=preview

**Comment by icfantv** [comment]
@bademux, please do not comment on old and closed issues.  Doing so is a really good way for us to miss it.

If you think you've found a bug, please follow [these instructions](https://github.com/angular-ui/bootstrap/blob/master/CONTRIBUTING.md#you-think-youve-found-a-bug).

Unfortunately, if you think you have found a bug and do not follow the above instructions within a reasonable amount of time, we will close the issue due to lack of activity.

**Comment by wes-r** [comment]
@icfantv Can you please re-open this?

https://github.com/angular-ui/bootstrap/commit/68afc4c63e791e86de7fe59be9bc8fb7e68243f6 should not have closed this as that adds `on-open-focus` and not the proposed `open-on-focus` as outlined in this issue. The former allows for "disabling of focus of popup after datepicker popup is opened" while the later would allow for the previous behavior on input elements (where focusing would open the popup).

Presently, the best solution is as @peterslivka said: `<input datepicker-popup is-open="focus" ng-focus="focus=true" />` which is gross o_O.
